extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProvidedChecksConfig.java

```diff
@@ -0,0 +1,186 @@
+package io.quarkus.smallrye.health.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocSection;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
```

Comment:
Hmmm, I'm under the impression that you will want to change most of these values at runtime?

Reply:
Meaning it should be possible to add and remove (and reconfigure) these health checks dynamically? I don't think that would be very useful.

Reply:
@gsmet what kind of scenarios do you have in mind?

I am drawing a blank on trying to imagine why the properties should be runtime instead of build time.

Reply:
Well, don't you expect `double maxPercentage;` to be deployment-specific? I certainly do.

Same for `Optional<String> host;` in `InetAddressHealthCheckConfig` and others.

Most of the checks are deployment-specific.

Reply:
Wouldn't moving them to runtime create a problem, because we wouldn't be able to tell at build-time whether we should create the health-check producers and the synthetic bean containing the configuration for them?

Reply:
Yes, sure, you will have this issue except if you have some sort of flags to enable the features.

As it is now, I can't see it being useful as you will have to rebuild your app for all the environments. And you wouldn't be able to tweak things without rebuilding your app.

Reply:
I don't quite like the idea that we would have to create all these health-check beans and a configuration bean always, even when they will be unused at runtime.
I could imagine partially solving that by adding an extra build-time property which "enables" provided health checks (creates *all* the necessary beans) and then we would somehow "disable" the beans at runtime if needed (not yet sure how, maybe it would need a mechanism in smallrye-health). But it would be one extra thing to be configured by the user, and we would probably still need to register all the beans even if only some (or none) of them will be eventually used.